src/physics/em/detail/BetheHeitler.cu

```diff
@@ -0,0 +1,94 @@
+//---------------------------------*-CUDA-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file BetheHeitler.cu
+//---------------------------------------------------------------------------//
+#include "BetheHeitler.hh"
+
+#include "base/KernelParamCalculator.cuda.hh"
+#include "physics/base/SecondaryAllocatorView.hh"
+#include "physics/base/ModelInterface.hh"
+#include "physics/base/ParticleTrackView.hh"
+#include "physics/base/PhysicsTrackView.hh"
+#include "physics/material/MaterialTrackView.hh"
+#include "random/cuda/RngEngine.hh"
+#include "BetheHeitlerInteractor.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+namespace
+{
+//---------------------------------------------------------------------------//
+// KERNELS
+//---------------------------------------------------------------------------//
+/*!
+ * Interact using the Bethe-Heitler model on applicable tracks.
+ */
+__global__ void bethe_heitler_interact_kernel(const BetheHeitlerPointers  bh,
+                                              const ModelInteractPointers ptrs)
+{
+    auto tid = celeritas::KernelParamCalculator::thread_id();
+    if (tid.get() >= ptrs.states.size())
+        return;
+
+    SecondaryAllocatorView allocate_secondaries(ptrs.secondaries);
+    ParticleTrackView particle(ptrs.params.particle, ptrs.states.particle, tid);
+
+    // Setup for ElementView access
+    MaterialTrackView material(ptrs.params.material, ptrs.states.material, tid);
+    // Cache the associated MaterialView as function calls to MaterialTrackView
+    // are expensive
+    MaterialView material_view = material.material_view();
+    CELER_ENSURE(material_view.num_elements() == 1);
+
+    PhysicsTrackView physics(ptrs.params.physics,
+                             ptrs.states.physics,
+                             particle.def_id(),
+                             MaterialDefId{},
+                             tid);
+
+    // This interaction only applies if the Bethe-Heitler model was selected
+    if (physics.model_id() != bh.model_id)
+        return;
+
+    BetheHeitlerInteractor interact(
+        bh,
+        particle,
+        ptrs.states.direction[tid.get()],
+        allocate_secondaries,
+        material_view.element_view(celeritas::ElementComponentId{0}));
```

Comment:
Can you add:
```c++
CELER_ASSERT(material_view.num_elements() == 1);
```
before this since for the moment we're hard-coding to a single element?

Reply:
This check is already there:
https://github.com/celeritas-project/celeritas/pull/111/files/42268f5c294a3d09c75d958e066f1441dcd9c2a6#diff-e45c92d92a135b17f866c2d59d144dfed3e8edd22b9c7fb8b5f20c95e7be602aR46

You want it moved?

Reply:
Sorry, I had used `CELER_ENSURE` instead of `CELER_ASSERT`. Updated.

Reply:
Oh duh, I missed it there. Yeah, I think the assertion is better closer to where the underlying assumption is made (not that the value of the assert will change between here and there, but it makes the reasoning for the assertion clearer).

Can you also change to `CELER_ASSERT` (internal assertion)? The `ENSURE` is for describing postconditions of a function.

Reply:
Done.